packages/react/src/BlockNoteTheme.ts

```diff
@@ -1,117 +1,160 @@
+import { CSSObject, MantineThemeOverride } from "@mantine/core";
+import { darkDefaultTheme, lightDefaultTheme } from "./defaultTheme";
+import { blockStyles } from "@blocknote/core";
 
-type ColorScheme = [
-  string,
-  string,
-  string,
-  string,
-  string,
-  string,
-  string,
-  string,
-  string,
-  string
-];
+export type CombinedColor = {
+  text: string;
+  background: string;
+};
+
+export type ColorScheme = {
+  editor: CombinedColor;
+  menu: CombinedColor;
+  tooltip: CombinedColor;
+  hovered: CombinedColor;
+  selected: CombinedColor;
+  disabled: CombinedColor;
+  shadow: string;
+  border: string;
+  sideMenu: string;
+  highlightColors: {
+    gray: CombinedColor;
+    brown: CombinedColor;
+    red: CombinedColor;
+    orange: CombinedColor;
+    yellow: CombinedColor;
+    green: CombinedColor;
+    blue: CombinedColor;
+    purple: CombinedColor;
+    pink: CombinedColor;
+  };
+};
+
+export type Theme = {
+  colors: ColorScheme;
+  borderRadius: number;
+  fontFamily: string;
+};
+
+export type BlockNoteComponentStyles = Partial<{
+  // Block Side Menu items, Formatting Toolbar buttons
+  ActionIcon: CSSObject;
+  // Slash Menu, Formatting Toolbar dropdown, color picker dropdown
+  Menu: CSSObject;
+  // Icon in the color picker dropdown (Formatting Toolbar & Drag Handle Menu)
+  ColorIcon: CSSObject;
+  DragHandleMenu: CSSObject;
+  // Menu to edit hyperlinks (in Formatting Toolbar & Hyperlink Toolbar)
+  EditHyperlinkMenu: CSSObject;
+  Editor: CSSObject;
+  // Wraps Formatting Toolbar & Hyperlink Toolbar
+  Toolbar: CSSObject;
+  // Appears on hover for Formatting Toolbar & Hyperlink Toolbar buttons
+  Tooltip: CSSObject;
+  SlashMenu: CSSObject;
+  SideMenu: CSSObject;
+}>;
+
+function shouldUseDarkTheme(
+  theme: Partial<Theme | { light: Partial<Theme>; dark: Partial<Theme> }>
+) {
+  const browserUsesDarkTheme = window.matchMedia(
+    "(prefers-color-scheme: dark)"
+  ).matches;
+
+  if ("light" in theme && "dark" in theme) {
+    return browserUsesDarkTheme;
+  }
 
-export const blockNoteColorScheme: ColorScheme = [
-  "#FFFFFF",
-  "#EFEFEF",
-  "#CFCFCF",
-  "#AFAFAF",
-  "#7F7F7F",
-  "#3F3F3F",
-  "#1F1F1F",
-  "#161616",
-  "#0F0F0F",
-  "#000000",
-];
+  if ("light" in theme) {
+    return false;
+  }
 
-export const getBlockNoteTheme = (
-  useDarkTheme: boolean = false
+  if ("dark" in theme) {
+    return true;
+  }
+
+  return browserUsesDarkTheme;
+}
+
+export const blockNoteToMantineTheme = (
+  theme: Partial<Theme | { light: Partial<Theme>; dark: Partial<Theme> }> = {
+    light: lightDefaultTheme,
+    dark: darkDefaultTheme,
+  },
+  componentStyles?: BlockNoteComponentStyles
 ): MantineThemeOverride => {
-  const boxShadow = `0px 4px 8px ${
-    useDarkTheme ? blockNoteColorScheme[8] : blockNoteColorScheme[2]
-  }, 0px 0px 1px ${
-    useDarkTheme ? blockNoteColorScheme[6] : blockNoteColorScheme[1]
-  }`;
+  const fullTheme: Theme = shouldUseDarkTheme(theme)
```

Comment:
This code and `shouldUseDarkTheme` is difficult to read. To fix: first check if { light:..., dark: ... } was passed, or whether a single theme was passed.

i.e.:

```
if ('light' in theme && 'dark' in theme) {
    theme = browserWantsDarkTheme() ? theme.dark : theme.light;
}
```